### PR TITLE
fix(ComponentForm): infinite rendering because of new Map ref everytime

### PR DESCRIPTION
**Pull request by jsomsanith-tlnd**
**What is the problem this PR is trying to solve?**
componentDidUpdate set states when properties have changed. But the comparison is ko when there is no properties passed, as it takes a new Map by default.

**What is the chosen solution to this problem?**
Compare the properties properly without default values to check if they have changed

**Please check if the PR fulfills these requirements**

- [x] The PR have used `yarn changeset` to a request a release from the CI if wanted.
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

**Review by gjou-tlnd — APPROVED** [review]

**Review by romainseb — APPROVED** [review]

**Review by jmfrancois — APPROVED** [review]

**Review by jmfrancois — CHANGES_REQUESTED** [review]
datalist seems broken http://3865.talend-playground.surge.sh/playground/

File: packages/containers/src/ComponentForm/ComponentForm.component.js [hunk 1]
```diff
@@ -102,9 +102,9 @@ export class TCompForm extends React.Component {
 	}
 
 	componentDidUpdate(prevProps) {
-		const nextProperties = this.props.state.get('properties', Map());
```
Comment (sgendre) [inline packages/containers/src/ComponentForm/ComponentForm.component.js:105]: extracting the allocation outside of the function would also work and bring less noise isn't it?

Comment (jsomsanith-tlnd) [reply packages/containers/src/ComponentForm/ComponentForm.component.js:105]: what ?

**Review by jmfrancois — APPROVED** [review]
datalist use triggers sorry

**Pull request merged**